File: backend/src/main/java/com/safetunnel/backend/vpn/session/VpnSessionRepository.java
package com.safetunnel.backend.vpn.session;

import com.safetunnel.backend.users.VpnSession;
import org.springframework.data.jpa.repository.JpaRepository;
import java.util.List;
import java.util.UUID;

public interface VpnSessionRepository extends JpaRepository<VpnSession, UUID> {
    List<VpnSession> findByUserIdOrderByConnectedAtDesc(UUID userId);
}

File: backend/src/main/java/com/safetunnel/backend/vpn/session/VpnSessionRepository.kt
package com.safetunnel.backend.vpn.session

import com.safetunnel.backend.users.VpnSession
import org.springframework.data.jpa.repository.JpaRepository
import java.util.UUID

interface VpnSessionRepository : JpaRepository<VpnSession, UUID> {
    fun findByUserIdOrderByConnectedAtDesc(userId: UUID): List<VpnSession>
}